crates/runtime/src/dataconnector/graphql.rs

```diff
@@ -245,7 +254,47 @@ fn unnest_json_object(unnest_parameters: &UnnestParameters, object: &Value) -> R
 
             // add the staged additions back to the root object
             for (key, value) in additions {
-                new_object.insert(key, value);
+                let new_key = match unnest_parameters.overwrite_behavior {
+                    OverwriteBehavior::Error => {
+                        if new_object.contains_key(&key) {
+                            return Err(Error::InvalidObjectAccess {
+                                message: format!(
+                                    "Column {key} already exists in the object and overwrite is set to error."
+                                ),
+                            });
+                        }
+
+                        key.clone()
+                    }
+                    OverwriteBehavior::Overwrite => key.clone(),
+                    OverwriteBehavior::Rename => {
+                        // if the key already exists, we need to rename it
+                        let mut new_key = key.clone();
+                        let counter = key_counter
+                            .entry(key.clone())
+                            .and_modify(|e| *e += 1)
+                            .or_insert(1);
+
+                        if new_object.contains_key(&key) {
+                            let Some(existing_value) = new_object.remove(&key) else {
+                                return Err(Error::InvalidObjectAccess {
+                                    message: format!("Column {key} already exists in the object and overwrite is set to rename, but the key does not exist."),
+                                });
+                            };
+
+                            new_object.insert(format!("{key}_{counter}"), existing_value);
```

Comment:
In theory, `counter` should always be 1 here right?

Reply:
That's correct, could probably rewrite to just set it to 1 in the format